src/formatDate.js

```diff
@@ -9,6 +9,62 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const outputNeeded = {
+    day: '',
+    month: '',
+    year: '',
+    separator: '',
+  };
+  const separatedDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  // check inData and start to build object for the result creating
+  for (let i = 0; i < fromFormat.length; i++) {
+    const item = fromFormat[i];
+
+    switch (item[0].toLowerCase()) {
+      case 'd':
+        outputNeeded.day = `${separatedDate[i]}`;
+        break;
+      case 'm':
+        outputNeeded.month = `${separatedDate[i]}`;
+        break;
+      case 'y':
+        const yearIn =
+          item.length === 2 && Number(separatedDate[i]) < 30
+            ? `20${separatedDate[i].slice(-2)}`
+            : `19${separatedDate[i].slice(-2)}`;
+
+        outputNeeded.year = item.length === 2 ? yearIn : `${separatedDate[i]}`;
+        break;
+      default:
+        outputNeeded.separator = `${toFormat[i]}`;
```

Comment:
There is no need to use template literals here since `toFormat[i]` is already a string. You can simply assign `outputNeeded.separator = toFormat[toFormat.length - 1];` outside of the loop since the separator does not change per iteration.